tests/unit/test_mongod_helpers.py

```diff
@@ -322,5 +323,32 @@ def test_check_replica_status_error(self, mock_client, client):
         for exception, expected_raise in PYMONGO_EXCEPTIONS:
             with self.assertRaises(expected_raise):
                 mock_client.admin.command.side_effect = exception
-                mongo.check_replica_status()
+                mongo.check_replica_status("1.1.1.1")
             mock_client.close.assert_called()
+
+    @patch("mongod_helpers.MongoDB.is_replica_ready", return_value=False)
+    @patch("pymongo.MongoClient")
+    @patch("mongod_helpers.MongoDB.client")
+    def test_not_all_replicas_ready(self, is_replica_ready, mock_client, client):
+        """Verifies that all_replicas_ready returns False when not all replicas are ready."""
+        # standard presets
+        config = MONGO_CONFIG.copy()
+        mongo = MongoDB(config)
+
+        # verify result of all replicas ready
+        self.assertEqual(mongo.all_replicas_ready(), False)
```

Comment:
should we have a test for all replicas being ready?

Reply:
Hi Shayan :D 

Great point, this case actually gets covered in our integration tests and since we are adopting a new way of testing where unit tests are meant to test situations which are not covered in the integration tests.